test/e2e/operator_test.go

```diff
@@ -3508,11 +3509,32 @@ func TestLoadBalancingAlgorithmUnsupportedConfigOverride(t *testing.T) {
 func TestUnsupportedConfigOverride(t *testing.T) {
 	t.Parallel()
 
+	var (
+		dcmDefaultValue         = ""
+		dcmUpdateValue          = "true"
+		dcmExpectedValue        = "true"
+		maxServersDefaultValue  = ""
+		maxServersUpdateValue   = "2"
+		maxServersExpectedValue = "" // max dynamic servers cannot be set if DCM is off
+	)
+
+	if dcmEnabled, err := isFeatureGateEnabled(features.FeatureGateIngressControllerDynamicConfigurationManager); err != nil {
+		t.Fatalf("failed to get dynamic config manager feature gate: %v", err)
+	} else if dcmEnabled {
+		dcmDefaultValue = "true"
+		dcmUpdateValue = "false"
+		dcmExpectedValue = ""
+		maxServersDefaultValue = "1"
+		maxServersUpdateValue = "2"
+		maxServersExpectedValue = "2"
+	}
+
 	var tests = []struct {
-		name, unsupportedConfigOverride, env string
+		name, unsupportedConfigOverride, env, defaultValue, updateValue, expectedValue string
 	}{
-		{"dynamic-config-manager", "dynamicConfigManager", "ROUTER_HAPROXY_CONFIG_MANAGER"},
-		{"contstats", "contStats", "ROUTER_HAPROXY_CONTSTATS"},
+		{"dynamic-config-manager", "dynamicConfigManager", "ROUTER_HAPROXY_CONFIG_MANAGER", dcmDefaultValue, dcmUpdateValue, dcmExpectedValue},
+		{"contstats", "contStats", "ROUTER_HAPROXY_CONTSTATS", "", "true", "true"},
+		{"max-dynamic-servers", "maxDynamicServers", "ROUTER_MAX_DYNAMIC_SERVERS", maxServersDefaultValue, maxServersUpdateValue, maxServersExpectedValue},
```

Comment:
nit: We do have variable names for many of these hard-coded values.

Reply:
Also, there's no test for the route blueprint default.

Reply:
> Also, there's no test for the route blueprint default.

Right, the test was made the way that only 1 variable is expected to be checked. I didn't have a goal to fully cover all configuration options in this e2e test, I just wanted to adapt the test to work with two new config overrides. The route blueprint configuration is well covered in the unit tests.